### ISSUE TITLE
S3 ViRGE does not work properly with Win95

### ISSUE THREAD

**Issue by DSN9990** [joncampbell123/dosbox-x#2447]
I can't get the recently added S3 ViRGE to work with Win95. Win95 throws up an error message that the display adapter is not configured correctly every time I start up Windows, and it stucks at 640x480 4bpp (16-color). @joncampbell123 Can you check this out? I know the ViRGE is experimental at this time, but I want to give it a try, though.

**Comment by joncampbell123** [comment]
Are you using machine=svga_s3virge?

The PCI card and CRTC registers need to report model and revision numbers that indicate a ViRGE.

**Comment by joncampbell123** [comment]
What *should* happen, if hardware acceleration is enabled on your desktop, is that Windows 95/98/3.1 should boot up with a nearly blank screen that is your desktop because the S3D acceleration they use is not yet implemented.

**Comment by joncampbell123** [comment]
You'll need to obtain the ViRGE drivers from 3rd party sites like the Vogons library because Windows 3.1/95 do not have built-in drivers for those cards.

**Comment by DSN9990** [comment]
@joncampbell123 Yes, I am using the setting machine=svga_s3virge as you said. As for the drivers, I'm using Windows 95C (OSR2.5) which does appear to have drivers for the ViRGE. Here is a screenshot:

![virgedrivers](https://user-images.githubusercontent.com/69558812/114835312-d135df00-9da7-11eb-9817-07359753874d.PNG)

The only issue is that I always get this error at startup saying it is not set up properly:

![virgeincorrectconfig](https://user-images.githubusercontent.com/69558812/114835721-3e497480-9da8-11eb-9d9a-02f1f3c47669.PNG)

**Comment by DSN9990** [comment]
One more question, machine=svga_s3virge is reported as a ViRGE 325 or something like DX, GX, etc?

**Comment by joncampbell123** [comment]
machine=svga_s3virge should be the basic model, there is a machine=svga_s3virgevx setting as well.

**Comment by DSN9990** [comment]
@Wengier Can you compile the latest code and give me a VS2019 (not MinGW) x64 SDL1 build?

**Comment by joncampbell123** [comment]
I am posting a mid-month beta release now, for testing.

**Comment by DSN9990** [comment]
@Wengier Can you give me a MinGW x64 SDL2 build? It seems that I got full 80-bit precision rendering issues on the VS2019 build.

**Comment by Wengier** [comment]
@DSN9990 The official MinGW builds for DOSBox-X 0.83.13 beta may be available shortly, but in case you want it now you can try this:

* [dosbox-x-mingw64-sdl2.zip](http://cndos.fam.cx/dosware/dosbox-x-mingw64-sdl2.zip)

**Comment by Diatonator** [comment]
I tested this card on my Win98. Windows installed the drivers. I tried to set 16bit/640x480 and got this surreal experience after reboot:
[Win98_S3Virge_VideoCapture.zip](https://github.com/joncampbell123/dosbox-x/files/6329681/Win98_S3Virge_VideoCapture.zip)
Safe mode 16colors/640x480 seems to be more or less working.

**Comment by joncampbell123** [comment]
S3 ViRGE at this time is very experimental. Windows drivers use S3D XGA registers that are not yet implemented.

**Comment by DSN9990** [comment]
@joncampbell123 How can I disable hardware acceleration in the System settings?

**Comment by DSN9990** [comment]
@Wengier Give me a updated MinGW 64 SDL2 build that include the latest commits, such as the fix for #2451.

**Comment by Diatonator** [comment]
> @joncampbell123 How can I disable hardware acceleration in the System settings?

Sorry my OS isn't in EN language, but as the process is not much complicated you will probably get the idea how to do it:
![image](https://user-images.githubusercontent.com/75184268/115154758-282c0480-a085-11eb-8757-8169ba95d8b1.png)
![image](https://user-images.githubusercontent.com/75184268/115154806-66c1bf00-a085-11eb-964e-892d536b7872.png)
![image](https://user-images.githubusercontent.com/75184268/115154826-81943380-a085-11eb-8434-9907029d344a.png)

I tried this on my Win98 - now the OS itself works OK visially.

**Comment by DSN9990** [comment]
@Wengier Like I said above, can you give me a updated MinGW x64 SDL2 build, please?

**Comment by Wengier** [comment]
Here: [dosbox-x-mingw64-sdl2.zip](http://cndos.fam.cx/dosware/dosbox-x-mingw64-sdl2.zip)
